jest.config.base.js

```diff
@@ -1,51 +1,147 @@
 const path = require('path')
```

Comment:
maybe the file name of this one should reflect that it's no longer exporting a jest config, but a config creator? Rename to `createJestConfig.ts` or something?